libs/execution/include/hpx/execution/executors/parallel_executor_aggregated.hpp

```diff
@@ -125,13 +128,13 @@ namespace hpx { namespace parallel { namespace execution {
                 lcos::local::latch l(size);
                 if (hpx::detail::has_async_policy(Policy{}))
                 {
-                    spawn_hierarchical(l, size, num_tasks, std::forward<F>(f),
+                    spawn_hierarchical(l, size, num_tasks, f,
```

Comment:
why do you not forward `f` here?

Reply:
I ran into dangling references. We need to revisit argument handling inside asynchronous algorithms thoroughly. But that's for a separate PR, I think. For now, this change fixed my issues.